tensorflow/core/grappler/grappler_item.cc

```diff
@@ -208,8 +208,16 @@ std::vector<const NodeDef*> ComputeTransitiveFanin(
   return result;
 }
 
+std::vector<const NodeDef*> ComputeTransitiveFanin(
+    const GraphDef& graph, const std::vector<string>& terminal_nodes, 
+    bool* ill_formed) {
+    std::unordered_map<string, const NodeDef*> name_to_fanin_node;
+    return ComputeTransitiveFanin(graph , terminal_nodes, name_to_fanin_node, ill_formed);
+}
+
 std::vector<const NodeDef*> ComputeTransitiveFanin(
     const GraphDef& graph, const std::vector<string>& terminal_nodes,
+    std::unordered_map<string, const NodeDef*>& name_to_fanin_node,
```

Comment:
Can you use pointer instead of reference? It's kinda convention in tensorflow code that input is const ref, and output or input/output is pointer.

Reply:
done